### PR TITLE
avm2: Implement flash.utils.Timer and associated events

### PR DESCRIPTION
**Pull request by Aaron1011**
The current 'setInterval/setTimeout' implementation is
moved to 'core/src/timers.rs', and now works with both
AVM1 and AVM2 objects. The `flash.utils.Timer` class is implemented
mostly in ActionScript, with minimal modifications to the actual
Ruffle timer code.

### CODE REVIEW

File: core/src/avm1/globals.rs [hunk 1]
```diff
@@ -290,36 +290,43 @@ pub fn create_timer<'gc>(
...
-    use crate::avm1::timer::TimerCallback;
-    let (callback, i) = match args.get(0) {
-        Some(Value::Object(o)) if o.as_executable().is_some() => (TimerCallback::Function(*o), 1),
+    use crate::timer::TimerCallback;
+
+    let (callback, interval) = match args.get(0) {
+        Some(Value::Object(o)) if o.as_executable().is_some() => (
+            TimerCallback::Avm1Function {
+                func: *o,
+                params: args.get(2..).map(|s| s.to_vec()).unwrap_or_default(),
```
Comment (Herschel) [inline core/src/avm1/globals.rs:299]: `args.get(2..).unwrap_or_default().to_vec()` seems a little clearer

File: core/src/avm2/globals/flash/utils/Timer.as [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			}
+		}
+
+		public function get repeatCount(): int {
+			return this._repeatCount;
+		}
+
+		public function set repeatCount(value:int): void {
+			this._repeatCount = value;
+		}
```
[the remark this answers is not in the document]
Comment (Aaron1011) [reply core/src/avm2/globals/flash/utils/Timer.as]: Nice catch - it looks like the behavior is different when setting `repeatCount` *inside* an event handler vs *outside*. I've added tests for both cases.

**Review by Herschel — APPROVED** [review]
Thank you!

**Comment by torokati44** [comment]
Just wanted to say that I'm super stoked that this is finally in place! \o/

**Pull request merged**